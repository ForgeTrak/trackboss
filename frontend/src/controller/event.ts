import { generateHeaders, getEventMonthDay, getTimeOfDay } from './utils';
import {
    DeleteEventResponse,
    GetEventListResponse,
    GetEventResponse,
    PatchEventRequest,
    PatchEventResponse,
    PostNewEventRequest,
    PostNewEventResponse,
    Event,
} from '../../../src/typedefs/event';
import { ErrorResponse } from '../../../src/typedefs/errorResponse';

function isEventList(res: Event[] | ErrorResponse): res is Event[] {
    return (res as Event[]) !== undefined;
}

export async function createEvent(token: string, eventData: PostNewEventRequest): Promise<PostNewEventResponse> {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/event/new`, {
        method: 'POST',
        mode: 'no-cors',
        headers: generateHeaders(token),
        body: JSON.stringify(eventData),
    });
    return response.json();
}

// TODO: this is a mocked response for frontend development, replace once API is complete
export async function makeEvent(name: string, description: string, start: Date, end: Date, typeId: number) {
    // eslint-disable-next-line no-console
    console.debug({
        date: start,
        eventTypeId: typeId,
        eventName: name,
        eventDescription: description,
    });
}

// TODO: this is a mocked response for frontend development, replace once API is completed
export async function getCalendarEventList() {
    // const upcomingEvents: GetEventListResponse = await getEventList('TestToken', todayString);
    // if ('reason' in upcomingEvents) {
    //     console.log('TODO: error handling?');
    // } else {
    //     convertEventsToCalendarFormat(upcomingEvents);
    //     return upcomingEvents;
    // }
    const upcomingEvents = [
        {
            title: 'Work Day',
            start: new Date('2022-02-08T12:00:00'),
            end: new Date('2022-02-09T15:50:00'),
            workPoints: 3,
            type: 'job',
        },
        {
            title: 'Race Day',
            start: new Date('2022-02-11T03:10:00'),
            end: new Date('2022-02-12T14:10:00'),
            type: 'race',
        },
        {
            title: 'Work Day',
            start: new Date('2022-02-11T03:10:00'),
            end: new Date('2022-02-12T14:10:00'),
            workPoints: 3,
            type: 'job',
        },
        {
            title: 'Work Day',
            start: new Date('2022-02-11T03:10:00'),
            end: new Date('2022-02-12T14:10:00'),
            workPoints: 3,
            type: 'job',
        },
        {
            title: 'Work Day',
            start: new Date('2022-02-11T03:10:00'),
            end: new Date('2022-02-12T14:10:00'),
            workPoints: 3,
            type: 'job',
        },
        {
            title: 'Meeting',
            start: new Date('2022-02-22T15:30:00'),
            end: new Date('2022-02-22T12:00:00'),
            type: 'meeting',
        },
        {
            title: 'Some other category',
            start: new Date('2022-02-22T00:10:00'),
            end: new Date('2022-02-23T00:10:00'),
            type: 'other',
        },
    ];

    return upcomingEvents;
}

export async function getEventList(token: string, listType?: string): Promise<GetEventListResponse> {
    if (listType) {
        const response = await fetch(`${process.env.REACT_APP_API_URL}/api/event/list`, {
            method: 'GET',
            mode: 'cors',
            headers: generateHeaders(token, listType),
        });
        return response.json();
    }
    // else
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/event/list`, {
        method: 'GET',
        mode: 'cors',
        headers: generateHeaders(token),
    });
    return response.json();
}

export async function getEventCardProps(token: string, listType: string) {
    const upcomingEvents = await getEventList(token, listType);

    if (isEventList(upcomingEvents)) {
        // + symbol here converts the dates to numbers, to allow for arithmetic comparison
        upcomingEvents.sort((e1: Event, e2: Event) => +new Date(e1.start) - +new Date(e2.start));
        const formattedEventDate = getEventMonthDay(upcomingEvents[0].start);
        const formattedEventTime = getTimeOfDay(upcomingEvents[0].start);
        return { title: upcomingEvents[0].title, start: formattedEventDate, time: formattedEventTime };
    }

    // else
    return undefined;
}

export async function getEvent(token: string, eventID: number): Promise<GetEventResponse> {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/event/${eventID}`, {
        method: 'GET',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}

export async function updateEvent(
    token: string,
    eventID: number,
    eventData: PatchEventRequest,
): Promise<PatchEventResponse> {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/event/${eventID}`, {
        method: 'PATCH',
        mode: 'no-cors',
        headers: generateHeaders(token),
        body: JSON.stringify(eventData),
    });
    return response.json();
}

export async function deleteEvent(token: string, eventID: number): Promise<DeleteEventResponse> {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/event/${eventID}`, {
        method: 'DELETE',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}
