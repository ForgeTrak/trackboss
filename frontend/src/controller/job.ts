import generateHeaders from './utils';

export async function createJob(token: string, jobData: object) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/job/new`, {
        method: 'POST',
        mode: 'no-cors',
        headers: generateHeaders(token),
        body: JSON.stringify(jobData),
    });
    return response.json();
}

export async function getJobList(token: string, queryType?: string, filterType?: string) {
    if (queryType && filterType) {
        const response = await fetch(
            `${process.env.REACT_APP_API_URL}/api/job/list?${queryType}=${filterType}`, {
                method: 'GET',
                mode: 'no-cors',
                headers: generateHeaders(token),
            },
        );
        return response.json();
    }
    // // else
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/job/list`, {
        method: 'GET',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}

export async function getJob(token: string, jobID: number) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/job/${jobID}`, {
        method: 'GET',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}

export async function updateJob(token: string, jobID: number, jobData: object) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/job/${jobID}`, {
        method: 'PATCH',
        mode: 'no-cors',
        headers: generateHeaders(token),
        body: JSON.stringify(jobData),
    });
    return response.json();
}

export async function cloneJob(token: string, jobID: number) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/job/${jobID}`, {
        method: 'POST',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}

export async function deleteJob(token: string, jobID: number) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/job/${jobID}`, {
        method: 'DELETE',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}
