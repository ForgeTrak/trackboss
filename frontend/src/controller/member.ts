import generateHeaders from "./utils";


export async function createMember(token: string, memberData: object) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/new`, {
        method: 'POST',
        mode: 'no-cors',
        headers: generateHeaders(token),
        body: JSON.stringify(memberData),
    });
    return response.json();
}

export async function getMember(token: string, memberId: number) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/${memberId}`, {
        method: 'GET',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}

export async function getMemberList(token: string, listType?: string) {
    if (listType) {
        const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/list?status=${listType}`, {
            method: 'GET',
            mode: 'no-cors',
            headers: generateHeaders(token),
        });
        return response.json();
    }
    // else
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/list`, {
        method: 'GET',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}

export async function updateMember(token: string, memberID: number, memberData: object) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/${memberID}`, {
        method: 'PATCH',
        mode: 'no-cors',
        headers: generateHeaders(token),
        body: JSON.stringify(memberData),
    });
    return response.json();
}