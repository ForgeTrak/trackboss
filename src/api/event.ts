import { format } from 'date-fns';
import { Request, Response, Router } from 'express';
import {
    deleteEvent, getClosestEvent, getEvent, getEventList, getRelatedEvents,
    insertEvent, patchEvent,
} from '../database/event';
import {
    DeleteEventResponse,
    Event,
    GetEventListResponse,
    GetEventResponse,
    PatchEventResponse,
    PostNewEventResponse,
} from '../typedefs/event';
import { checkHeader, verify } from '../util/auth';
import logger from '../logger';

const event = Router();

event.post('/new', async (req: Request, res: Response) => {
    const { authorization } = req.headers;
    let response: PostNewEventResponse;
    const headerCheck = checkHeader(authorization);
    if (!headerCheck.valid) {
        res.status(401);
        response = { reason: headerCheck.reason };
    } else {
        try {
            await verify(headerCheck.token, 'Admin');
            const insertId = await insertEvent(req.body);
            response = await getEvent(insertId);
            const relatedEvents = await getRelatedEvents(response);
            // awaiting all this on purpose because a) it is fast and b) I want the UI to update when all of this
            // is done. We are in maÃ±uel transmission mode here.
            // eslint-disable-next-line no-restricted-syntax
            for (const childEvent of relatedEvents) {
                // eslint-disable-next-line no-await-in-loop
                await insertEvent(childEvent);
            }
            res.status(201);
        } catch (e: any) {
            logger.error(`Error at path ${req.path}`);
            logger.error(e);
            if (e.message === 'user input error') {
                res.status(400);
                response = { reason: 'bad request' };
            } else if (e.message === 'Authorization Failed') {
                res.status(401);
                response = { reason: 'not authorized' };
            } else if (e.message === 'Forbidden') {
                res.status(403);
                response = { reason: 'forbidden' };
            } else {
                res.status(500);
                response = { reason: 'internal server error' };
            }
        }
    }
    res.send(response);
});

event.get('/list', async (req: Request, res: Response) => {
    const { authorization } = req.headers;
    let response: GetEventListResponse;
    const headerCheck = checkHeader(authorization);
    const verifyDate = (dateString: string) => {
        let valid = true;
        if (dateString === '') {
            valid = true;
        }
        const year = Number(dateString.slice(0, 4));
        const month = Number(dateString.slice(4, 6));
        const day = Number(dateString.slice(6, 8));
        if (Number.isNaN(year) || Number.isNaN(month) || Number.isNaN(day)) {
            valid = false;
        }
        return { valid, date: new Date(year, month - 1, day) };
    };
    if (!headerCheck.valid) {
        res.status(401);
        response = { reason: headerCheck.reason };
    } else {
        try {
            await verify(headerCheck.token);
            const { range } = req.headers;
            let eventList: Event[];
            if (typeof range === 'undefined') {
                eventList = await getEventList();
                res.status(200);
            } else {
                const [start, end] = range.split('-');
                if (start === undefined || end === undefined) {
                    throw new Error('user input error');
                }
                const startData = verifyDate(start);
                const endData = verifyDate(end);
                if (!startData.valid || !endData.valid) {
                    throw new Error('user input error');
                } else {
                    if (start === '' && end !== '') {
                        eventList = await getEventList(
                            undefined,
                            `${format(endData.date, 'yyyy-MM-dd')}`,
                        );
                    } else if (end === '' && start !== '') {
                        eventList = await getEventList(`${format(startData.date, 'yyyy-MM-dd')}`);
                    } else {
                        eventList = await getEventList(
                            `${format(startData.date, 'yyyy-MM-dd')}`,
                            `${format(endData.date, 'yyyy-MM-dd')}`,
                        );
                    }
                    res.status(206);
                }
            }
            response = eventList;
        } catch (e: any) {
            logger.error(`Error at path ${req.path}`);
            logger.error(e);
            if (e.message === 'user input error') {
                res.status(400);
                response = { reason: 'bad request' };
            } else if (e.message === 'Authorization Failed') {
                res.status(401);
                response = { reason: 'not authorized' };
            } else {
                res.status(500);
                response = { reason: 'internal server error' };
            }
        }
    }
    res.send(response);
});

event.get('/:eventID', async (req: Request, res: Response) => {
    const { authorization } = req.headers;
    let response: GetEventResponse;
    const headerCheck = checkHeader(authorization);
    if (!headerCheck.valid) {
        res.status(401);
        response = { reason: headerCheck.reason };
    } else {
        try {
            await verify(headerCheck.token);
            const { eventID } = req.params;
            if (eventID === 'next') {
                response = await getClosestEvent();
            } else {
                const id = Number(eventID);
                if (Number.isNaN(id)) {
                    throw new Error('not found');
                }
                response = await getEvent(id);
            }
            res.status(200);
        } catch (e: any) {
            logger.error(`Error at path ${req.path}`);
            logger.error(e);
            if (e.message === 'not found') {
                res.status(404);
                response = { reason: 'not found' };
            } else if (e.message === 'Authorization Failed') {
                res.status(401);
                response = { reason: 'not authorized' };
            } else {
                res.status(500);
                response = { reason: 'internal server error' };
            }
        }
    }
    res.send(response);
});

event.patch('/:eventID', async (req: Request, res: Response) => {
    const { authorization } = req.headers;
    let response: PatchEventResponse;
    const headerCheck = checkHeader(authorization);
    if (!headerCheck.valid) {
        res.status(401);
        response = { reason: headerCheck.reason };
    } else {
        try {
            const { eventID } = req.params;
            const id = Number(eventID);
            if (Number.isNaN(id)) {
                throw new Error('not found');
            }
            await verify(headerCheck.token, 'Admin');
            await patchEvent(id, req.body);
            response = await getEvent(id);
            res.status(200);
        } catch (e: any) {
            logger.error(`Error at path ${req.path}`);
            logger.error(e);
            if (e.message === 'user input error') {
                res.status(400);
                response = { reason: 'bad request' };
            } else if (e.message === 'not found') {
                res.status(404);
                response = { reason: 'not found' };
            } else if (e.message === 'Authorization Failed') {
                res.status(401);
                response = { reason: 'not authorized' };
            } else if (e.message === 'Forbidden') {
                res.status(403);
                response = { reason: 'forbidden' };
            } else {
                res.status(500);
                response = { reason: 'internal server error' };
            }
        }
    }
    res.send(response);
});

event.delete('/:eventID', async (req: Request, res: Response) => {
    const { authorization } = req.headers;
    let response: DeleteEventResponse;
    const headerCheck = checkHeader(authorization);
    if (!headerCheck.valid) {
        res.status(401);
        response = { reason: headerCheck.reason };
    } else {
        try {
            const { eventID } = req.params;
            const id = Number(eventID);
            if (Number.isNaN(id)) {
                throw new Error('not found');
            }
            await verify(headerCheck.token, 'Admin');
            await deleteEvent(id);
            response = { eventId: id };
            res.status(200);
        } catch (e: any) {
            logger.error(`Error at path ${req.path}`);
            logger.error(e);
            if (e.message === 'not found') {
                res.status(404);
                response = { reason: 'not found' };
            } else if (e.message === 'Authorization Failed') {
                res.status(401);
                response = { reason: 'not authorized' };
            } else if (e.message === 'Forbidden') {
                res.status(403);
                response = { reason: 'forbidden' };
            } else {
                res.status(500);
                response = { reason: 'internal server error' };
            }
        }
    }
    res.send(response);
});

export default event;
