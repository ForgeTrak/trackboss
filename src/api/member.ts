import { Request, Response, Router } from 'express';

const member = Router();

member.post('/new', (req: Request, res: Response) => {
    res.status(501).send();
});

member.get('/list', (req: Request, res: Response) => {
    res.status(501).send();
});

member.get('/:memberID', (req: Request, res: Response) => {
    res.status(501).send();
});

member.patch('/:memberID', (req: Request, res: Response) => {
    res.status(501).send();
});

member.delete('/:memberID', (req: Request, res: Response) => {
    res.status(501).send();
});

export default member;
