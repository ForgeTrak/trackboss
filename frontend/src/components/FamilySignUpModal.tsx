// import React, { useEffect, useState } from 'react';
import React, { useContext } from 'react';
import {
    Modal,
    ModalOverlay,
    ModalContent,
    ModalCloseButton,
    Button,
    Divider,
    Heading,
    ModalFooter,
    SimpleGrid,

} from '@chakra-ui/react';
import moment from 'moment';
import { Member } from '../../../src/typedefs/member';
import { Job, PatchJobRequest } from '../../../src/typedefs/job';
import { UserContext } from '../contexts/UserContext';

interface modalProps {
  isOpen: boolean,
  onClose: () => void,
  familyMembers: Member[],
  selectedEvent: any, // TODO: bugfix needed, Event | Job breaks here
  // eslint-disable-next-line no-unused-vars
  signUpForJob: (patchInfo: { jobId: number; editedJob: PatchJobRequest; }) => void
}

export default function FamilySignUpModal(props: modalProps) {
    const { state } = useContext(UserContext);

    function isJob(selectedEvent: Event | Job): selectedEvent is Job {
        if ((selectedEvent as Job).jobId) {
            return true;
        }
        // else, its an Event
        return false;
    }

    async function generateJobSignUpPatch(memberId: number) {
        let editedJob: PatchJobRequest;
        if (isJob(props.selectedEvent) && memberId && state.user) {
            const { jobId } = props.selectedEvent;
            editedJob = {
                memberId,
                eventId: props.selectedEvent.eventId,
                jobTypeId: undefined,
                jobStartDate: moment(props.selectedEvent.start).toISOString(true).slice(0, -10),
                jobEndDate: moment(props.selectedEvent.end).toISOString(true).slice(0, -10),
                pointsAwarded: props.selectedEvent.pointsAwarded,
                verified: props.selectedEvent.verified,
                paid: props.selectedEvent.paid,
                modifiedBy: state.user.memberId,
            };
            return { jobId, editedJob };
        }
        return undefined;
    }

    return (
        <Modal isCentered size="xl" isOpen={props.isOpen} onClose={props.onClose}>
            <ModalOverlay />
            <ModalContent>
                <Heading pl={2} pr={2} textAlign="left">Select member to sign up</Heading>
                <Divider mb={5} />
                <ModalCloseButton />
                <SimpleGrid spacing={2} columns={3}>
                    {
                        props.familyMembers.map((member) => (
                            <Button
                                key={member.memberId}
                                onClick={
                                    async () => {
                                        const signUpPatch = await generateJobSignUpPatch(member.memberId);
                                        if (signUpPatch) {
                                            props.signUpForJob(signUpPatch);
                                        }
                                        props.onClose();
                                    }
                                }
                                m={3}
                                _hover={{ bg: 'orange.100' }}
                            >
                                {`${member.firstName} ${member.lastName}`}
                            </Button>
                        ))
                    }
                </SimpleGrid>
                <Divider />
                <ModalFooter>
                    <Button
                        variant="ghost"
                        mr={3}
                        size="sm"
                        onClick={
                            () => {
                                props.onClose();
                            }
                        }
                    >
                        Close
                    </Button>
                </ModalFooter>
            </ModalContent>
        </Modal>
    );
}
