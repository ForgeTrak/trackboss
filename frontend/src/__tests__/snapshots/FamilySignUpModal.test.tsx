import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import FamilySignUpModal from '../../components/FamilySignUpModal';
import { Member } from '../../../../src/typedefs/member';
import { PatchJobRequest } from '../../../../src/typedefs/job';
import { Event } from '../../../../src/typedefs/event';

function onClose() {
    // a void function
}

const memberFamily: Member[] = [
    {
        membershipId: 1,
        memberId: 1,
        membershipAdmin: 'true',
        active: true,
        memberType: 'member',
        firstName: 'test',
        lastName: 'member',
        phoneNumber: '1234',
        email: 'user@example.com',
        uuid: '',
        occupation: '',
        birthdate: '',
        dateJoined: '',
        address: '',
        city: '',
        state: '',
        zip: '',
        lastModifiedDate: '',
        lastModifiedBy: '',
    },
    {
        membershipId: 1,
        memberId: 2,
        membershipAdmin: 'string',
        active: true,
        memberType: 'member',
        firstName: 'Jon',
        lastName: 'Smith',
        phoneNumber: '1234',
        email: 'user@example.com',
        uuid: '',
        occupation: '',
        birthdate: '',
        dateJoined: '',
        address: '',
        city: '',
        state: '',
        zip: '',
        lastModifiedDate: '',
        lastModifiedBy: '',
    },
];

async function signUpForJob(patchInfo: { jobId: number, editedJob: PatchJobRequest }) {
    console.log(patchInfo);
}

const selectedEvent: Event = {
    eventId: 1,
    start: 'test',
    end: 'test',
    eventType: 'test',
    title: 'test',
    eventDescription: 'test',
};

// Necessary to mock the chakra-ui modal portal, so that the renderer can target the modal instead of the page
const divWithChildrenMock = (children: any, identifier: any) => <div data-testId={identifier}>{children}</div>;
jest.mock('@chakra-ui/react', () => (
    {
        ...jest.requireActual('@chakra-ui/react'),
        PortalManager: jest.fn(({ children }) => divWithChildrenMock(children, 'portal')),
    }
));

describe('sign up modal', () => {
    it('renders all props correctly', () => {
        const modal = render(
            <FamilySignUpModal
                isOpen
                onClose={onClose}
                familyMembers={memberFamily}
                selectedEvent={selectedEvent}
                signUpForJob={signUpForJob}
            />,
        );
        expect(modal).toMatchSnapshot();
    });
});
