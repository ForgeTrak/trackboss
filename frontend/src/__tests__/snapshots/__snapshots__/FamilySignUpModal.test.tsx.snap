// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign up modal renders all props correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div />
    <div
      class="chakra-portal"
    >
      <div
        class="chakra-modal__overlay css-13d1tkw"
        style="opacity: 0;"
      />
      <div
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="1"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div
          class="chakra-modal__content-container css-k2m3mz"
          tabindex="-1"
        >
          <section
            aria-modal="true"
            class="chakra-modal__content css-6yiygt"
            id="chakra-modal-1"
            role="dialog"
            style="opacity: 0; transform: scale(0.95) translateZ(0);"
            tabindex="-1"
          >
            <h2
              class="chakra-heading css-w5vb1z"
            >
              Select member to sign up
            </h2>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-vrx5g4"
            />
            <button
              aria-label="Close"
              class="chakra-modal__close-btn css-6sdaxy"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <div
              class="css-1kjn4rh"
            >
              <button
                class="chakra-button css-16tbipg"
                type="button"
              >
                test member
              </button>
              <button
                class="chakra-button css-16tbipg"
                type="button"
              >
                Jon Smith
              </button>
            </div>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-1upb9tn"
            />
            <footer
              class="chakra-modal__footer css-192qrng"
            >
              <button
                class="chakra-button css-1q0x5os"
                type="button"
              >
                Close
              </button>
            </footer>
          </section>
        </div>
      </div>
      <div
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
